src/v/redpanda/admin/api-doc/broker.json

```diff
@@ -440,6 +462,13 @@
                         "type": "partition_reconfiguration_status"
                     },
                     "description": "Array of partition reconfiguration statues"
+                },
+                "reallocation_failure_details": {
```

Comment:
Given we have this new field, I have an impression we no longer need the `allocation_failures` field. As long as a partition is listed in `reallocation_failure_details`, it will be listed in `allocation_failures` as well, right?

Reply:
Yes, exactly. I left it for the backward compatibility.

Reply:
Yes, exactly. I left it for the backward compatibility.

Reply:
That's fair. rpk-wise, we'll do this way, which must be backward compatible. 

```
if reallocation_failure_details is not null, list partitions with failure details
else if allocation_failures is not null, list partitions
else there's no failed partitions
```